### DISCUSSION TITLE
Support for the TemporalNetV2 Model?

### DISCUSSION THREAD

**Discussion by SSA4** [Mikubill/sd-webui-controlnet#1752] [Q&A] [answered]
Hi,
the new TemporalnetV2 model by CiaraRowles! was released about 1 month ago. But unlike the TemporalNet V1 model, this model still cannot be used in the WebUI and in Controlnet. But maybe I'm making a mistake when using it. The old version of the model was loaded as model without a preprocessor without any problems. If the Loopback was enabled the consistency of the image sequences (batch process) was significantly improved. 

At the moment there a two versions of the Temporalnet2 available. 
The original model by "CiaraRowles!" (12GB big .ckpt file) and a diffusers-compatible checkpoint by "wav" (1.45GB big .safetensor file).

The links to the models:
https://huggingface.co/CiaraRowles/TemporalNet2
https://huggingface.co/wav/TemporalNet2

I've tried a lot to load one of these models in the WebUI in Controlnet, but I get an error message with each of these models. Has anyone managed to load the model or do we have to wait for a compatible version of the model to be released?

**Comment by ljleb** [comment, marked as answer]
TemporalNet2 uses optical flow as control maps, where TN1 used the previous time image directly. It needs a proper preprocessor to work in the webui.

As of right now it is not supported. You can follow the model card in the official hugging face repository by CiaraRowles to run some tests through the web API IIUC. They mention a branch that supports the model in TemporalKit and I think they also have a patched version of this extension, which may all be worth a try.
